### DISCUSSION TITLE
how to configure rotational axis to AC not AB (default) for 5 axis

### DISCUSSION THREAD

**Discussion by yuuahmad** [grblHAL/core#419] [Q&A] [answered]
I have problems configuring grblhal for 5 axis properly and correctly.

by default, axes 4 and 5 are rotation axis A and rotation axis B. However, I want axes 4 and 5 to be A and C because in CAM and in simulation I use these axes.

I can control axis A, but I cannot control axis C

So far, I have succeeded in displaying the axis on the console correctly but cannot run the steppermotor with the jog command on the gcode sender. and simple gcode commands like G1C10F100 also won't start the motor

I have made changes to config.h nut_bolts.c nuts_bolts.h and config.h. For details, you can see my repo at 
[STM32F4xx yuuahmad](https://github.com/yuuahmad/STM32F4xx/tree/defaultAB) and [core yuuahmad](https://github.com/yuuahmad/core/tree/defaultAB)

Below is a screenshot of my progress so far
`[VER:1.1f.20231229:]
[OPT:VNMSL,100,1024,5,0]
[AXS:5:XYZAC]
[NEWOPT:ENUMS,RT+,ES,SED]
[SIGNALS:HSE]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[FREE MEMORY:103K]
[DRIVER:STM32F446]
[DRIVER VERSION:231228]
[BOARD:yuuahmad machine]
[AUX IO:2,4,0,0]
ok
$X
ok
$0=10.0
$1=25
$2=0
$3=0
$4=7
$5=31
$6=0
$9=1
$10=511
$11=0.010
$12=0.002
$13=0
$14=70
$15=0
$16=0
$17=0
$18=0
$19=0
$20=0
$21=0
$22=0
$23=0
$24=25.0
$25=500.0
$26=250
$27=1.000
$28=0.100
$29=0.0
$30=1000.000
$31=0.000
$32=0
$33=5000.0
$34=0.0
$35=0.0
$36=100.0
$37=0
$39=1
$40=0
$43=1
$44=4
$45=3
$46=0
$47=0
$48=0
$62=0
$63=3
$64=0
$65=0
$100=250.00000
$101=250.00000
$102=250.00000
$103=250.00000
$104=250.00000
$110=500.000
$111=500.000
$112=500.000
$113=500.000
$114=500.000
$120=10.000
$121=10.000
$122=10.000
$123=10.000
$124=10.000
$130=200.000
$131=200.000
$132=200.000
$133=200.000
$134=200.000
$341=0
$342=30.0
$343=25.0
$344=200.0
$345=200.0
$346=1
$370=0
$372=0
$376=0
$384=0
$398=100
$481=0
$484=1
$486=0
ok
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[G54:0.000,0.000,0.000,0.000,0.000]
[G55:0.000,0.000,0.000,0.000,0.000]
[G56:0.000,0.000,0.000,0.000,0.000]
[G57:0.000,0.000,0.000,0.000,0.000]
[G58:0.000,0.000,0.000,0.000,0.000]
[G59:0.000,0.000,0.000,0.000,0.000]
[G59.1:0.000,0.000,0.000,0.000,0.000]
[G59.2:0.000,0.000,0.000,0.000,0.000]
[G59.3:0.000,0.000,0.000,0.000,0.000]
[G28:0.000,0.000,0.000,0.000,0.000]
[G30:0.000,0.000,0.000,0.000,0.000]
[G92:0.000,0.000,0.000,0.000,0.000]
[TLO:0.000,0.000,0.000,0.000,0.000]
[PRB:0.000,0.000,0.000,0.000,0.000:0]
ok
<Idle|MPos:0.000,0.000,0.000,-1.900,0.000|Bf:100,1023|FS:0,0|Pn:|WCO:0.000,0.000,0.000,0.000,0.000>
$pins
[PIN:PB12,Emergency stop]
[PIN:PC8,Feed hold]
[PIN:PC11,Cycle start]
[PIN:PB7,Probe]
[PIN:PA5,X limit min]
[PIN:PB9,Y limit min]
[PIN:PC13,Z limit min]
[PIN:PB6,A limit min]
[PIN:PC14,C limit min]
[PIN:PA0,Aux input 0,P0]
[PIN:PA1,Aux input 1,P1]
[PIN:PA10,X step]
[PIN:PC1,Y step]
[PIN:PB8,Z step]
[PIN:PD2,A step]
[PIN:PB14,C step]
[PIN:PC2,X dir]
[PIN:PC0,Y dir]
[PIN:PC15,Z dir]
[PIN:PC12,A dir]
[PIN:PB15,C dir]
[PIN:PA14,X enable]
[PIN:PA14,Y enable]
[PIN:PA13,Z enable]
[PIN:PA14,A enable]
[PIN:PA14,C enable]
[PIN:PB2,Spindle on]
[PIN:PB1,Spindle direction]
[PIN:PC9,Flood]
[PIN:PA7,Mist]
[PIN:PB13,Aux out 0,P0]
[PIN:PB0,Aux out 1,P1]
[PIN:PA4,Aux out 2,P2]
[PIN:PA6,Aux out 3,P3]
[PIN:PA3,RX,Primary UART]
[PIN:PA2,TX,Primary UART]
[PIN:PC11,RX,UART2]
[PIN:PC10,TX,UART2]
[PIN:PC6,SCK]
[PIN:PC7,SDA]
[PIN:PA8,Spindle PWM]
ok
`

![Cuplikan layar 2024-01-03 155720](https://github.com/grblHAL/core/assets/22028802/b43082dc-3b1f-4d9a-9ea4-a79ee5f53304)

**Comment by terjeio** [comment, marked as answer]
The core does not support arbitrary naming of axes, they have to be consecutive. You may get around it by modifying the core by substituting C for B where needed. Another option would be to use a 6 axis map and set the stepper pins for B (motor 4) to the same (unused) pin.

**Reply by yuuahmad** [reply]
Thank you for the answer. I will try again and report the results later
